### PR TITLE
Creates a data summary component

### PR DESCRIPTION
**Pull request by darwinlopez**
Creates a data summary component - includes main data container and item container.

### CODE REVIEW

File: src/rb-data-summary/rb-data-summary-directive.js [hunk 1]
```diff
@@ -22,12 +26,19 @@ define([
...
+            template: template,
+            link: link
         };
+
+        function link (scope, elem, attr) {
+
+            if (angular.isDefined(attr.type)) {
+                elem.addClass('DataSummary--' + attr.type);
+            }
+        }
```
Comment (chrisvxd) [inline src/rb-data-summary/rb-data-summary-directive.js]: Why don't you not like the `ng-class` directive? I've noticed you often add classes this way.

File: test/unit/rb-data-summary/rb-data-summary-item.spec.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            $scope = _$rootScope_.$new({});
+            $compile = _$compile_;
+            rbDataSummaryItem = angular.element(template);
+            element = $compile(rbDataSummaryItem)($scope);
+            $scope.$apply();
+        }));
+
+        describe('attribute generation', function () {
+
+            it('should convert attributes on a rb-header to attributes on the generated header',
```
Comment (chrisvxd) [inline test/unit/rb-data-summary/rb-data-summary-item.spec.js]: This isn't `rb-header`

File: test/unit/rb-data-summary/rb-data-summary-item.spec.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        });
+
+        describe('rendering', function () {
+
+            it('should render with a "div" tagname', function () {
+                expect(element[0].tagName.toLowerCase()).toEqual('div');
+            });
+        });
+
+    });
```
Comment (chrisvxd) [inline test/unit/rb-data-summary/rb-data-summary-item.spec.js]: Shouldn't we be checking that `header` is rendered?

File: test/unit/rb-data-summary/rb-data-summary.spec.js [hunk 4]
```diff
@@ -19,8 +19,34 @@ define([
             $scope.$apply();
         }));
 
-        it('should do something', function () {
+        describe('attribute generation', function () {
+
+            it('should convert attributes on a rb-header to attributes on the generated header',
```
Comment (chrisvxd) [inline test/unit/rb-data-summary/rb-data-summary.spec.js]: Again, this isn't `rb-header`

File: src/rb-data-summary/rb-data-summary-directive.js (hunk 1 above)
Comment (darwinlopez) [inline src/rb-data-summary/rb-data-summary-directive.js]: Because this way is faster for angular and potentially no watcher is added

Comment (chrisvxd) [inline src/rb-data-summary/rb-data-summary-directive.js]: is there a watcher when using `ng-class="DataSummary--{{type}}"`?

Comment (darwinlopez) [inline src/rb-data-summary/rb-data-summary-directive.js]: yup

File: test/unit/rb-data-summary/rb-data-summary.spec.js (hunk 4 above)
Comment (darwinlopez) [inline test/unit/rb-data-summary/rb-data-summary.spec.js]: Done

File: test/unit/rb-data-summary/rb-data-summary-item.spec.js (hunk 3 above)
Comment (darwinlopez) [inline test/unit/rb-data-summary/rb-data-summary-item.spec.js]: Done

File: test/unit/rb-data-summary/rb-data-summary-item.spec.js (hunk 2 above)
Comment (darwinlopez) [inline test/unit/rb-data-summary/rb-data-summary-item.spec.js]: Done

**Pull request merged**